backend/handler/auth/middleware.py

```diff
@@ -107,7 +109,15 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
                 jwt_claims = self._validate_jwt_payload(jwt_payload)
                 scope["session"] = jwt_claims
                 initial_session_was_empty = False
-            except BadSignatureError:
+
+                # Check if the user exists
+                db_user_handler = DBUsersHandler()
+                user = db_user_handler.get_user_by_username(jwt_claims.get("sub"))
+                if not user:
+                    scope["session"] = {}
+                    initial_session_was_empty = True
```

Comment:
making a db request on every request is very expensive, and i think there's a better way to handle this...

Reply:
maybe we can make the db request only in the heartbeat instead on every request? that will reduce the database requests but will clear session when refreshing the page if the user no longer exists

Reply:
i tested it and it won't work, if any requests is made before the heartbeat it will crash